src/views/Play.vue

```diff
@@ -1,102 +0,0 @@
-<template>
-  <div class="home">
-    <h1 class="title">Kitten Compare</h1>
-    <h2 class="subtitle">Which one do you like best?</h2>
-    <Comparator :id1="this.currentRound.sample1.contenderId"
-                :id2="this.currentRound.sample2.contenderId"
-                :imageUrl1="this.currentRound.sample1.sampleUrl"
-                :imageUrl2="this.currentRound.sample2.sampleUrl"
-                v-on:winner="handleRoundResult"/>
-    <h2 class="subtitle" style="margin-bottom: 6px">The Current Rating:</h2>
-    <RankingStrip :contendersSortedByRating="allContendersSortedByTheirRating"/>
-  </div>
-</template>
-
-<script>
-import { mapGetters } from "vuex";
-// @ is an alias to /src
-import Comparator from "@/components/Comparator.vue";
-import RankingStrip from "@/components/RankingStrip.vue";
-import { twoDifferentRandomIdsInRange } from "@/util/IdGeneratorUtil.js";
-
-export default {
-  name: "home",
-  components: {
-    Comparator,
-    RankingStrip
-  },
-  data() {
-    return {
-      // Model for the current game round.
-      currentRound: this.roundModelForContenders(null, null)
-    };
-  },
-  computed: {
-    ...mapGetters([
-      "allContendersSortedByTheirRating",
-      "imageUrlForContender",
-      "currentContenderIdRange",
-      "gameServerEnabled",
-      "gameServerUrl"
-    ])
-  },
-  created() {
-    // @see: https://vuejs.org/v2/guide/instance.html#Instance-Lifecycle-Hooks.
-    if (this.gameServerEnabled) {
-      this.$store.dispatch("startContenderPolling");
-      // We wait for the list of contenders to change, because only then
-      // we can be sure that new data arrived from the server and start the next round.
-      const unwatch = this.$store.watch(
-        (state, getters) => getters.allContendersSortedByTheirRating,
-        () => {
-          this.startNextRound();
-          unwatch();
-        }
-      );
-    } else {
-      this.startNextRound();
-    }
-  },
-  methods: {
-    handleRoundResult(winnerId) {
-      const contenderId1 = this.currentRound.sample1.contenderId;
-      const contenderId2 = this.currentRound.sample2.contenderId;
-
-      const roundResult = {
-        contenderId1,
-        contenderId2,
-        winnerId
-      };
-
-      if (this.gameServerEnabled) {
-        this.$store.dispatch("publishRoundResult", roundResult);
-      } else {
-        this.$store.dispatch("updateRatingAction", roundResult);
-      }
-      this.startNextRound();
-    },
-    async startNextRound() {
-      if (this.gameServerEnabled) {
-        const serverResponse = await fetch(`${this.gameServerUrl}/round/new`);
```

Comment:
Sollte der fetch call innerhalb einer vuex action ablaufen? Vll über mapActions reinziehen und currentRound komplett aus einem Vuex getter beziehen?
alternativ evtl den store.dispatch für das fetch() awaiten, die actions returnen ein Promise.. <3 Vue <3 `await this.$store.dispatch(...);"

Reply:
Hm.. ich dachte bis dato es wäre best practise, nichtdeterministischen code mit side effects (z.b. remote status update) in actions auszulagern bzw. services, die von actions verwendet werden... und die views/pages/container components hängen nur vom state ab und rendern ebenfalls deterministisch basierend auf dem state (page reload = gleicher state = gleicher view state). macht auch das testen leichter.

Reply:
Das ist eine Auffassung, die viele eXXcellent-Kollegen teilen, aber imho. so nicht immer richtig ist. Es kommt eben darauf an, und genau diese Diskussion wollte ich an der Stelle anstoßen. 

Aus der Vuex-Doku:
Components Can Still Have Local State
Using Vuex doesn't mean you should put all the state in Vuex. Although putting more state into Vuex makes your state mutations more explicit and debuggable, sometimes it could also make the code more verbose and indirect. If a piece of state strictly belongs to a single component, it could be just fine leaving it as local state. You should weigh the trade-offs and make decisions that fit the development needs of your app.

Auch in der React-Doku werden API-Calls direkt in Components abgesetzt:
https://reactjs.org/docs/faq-ajax.html